File: POS/POS/InventoryEditCategoryAddForm.vb
Imports MySql.Data.MySqlClient

Public Class InventoryEditCategoryAddForm

    Dim conn As New MySqlConnection
    Dim connstring As String = "Server=localhost;User Id=root;Password=;Database=POS"

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        If TextBox1.Text = "" Then
            MsgBox("Please input name for new category")
            Exit Sub
        End If
        If InventoryEditCategoryForm.ListBox1.Items.Contains(Me.TextBox1.Text) Then
            MsgBox("Category already exists")
            Exit Sub
        End If
        Try
            conn.ConnectionString = connstring
            conn.Open()

            Dim comm As New MySqlCommand
            Dim query As String = "INSERT INTO productcategorytbl (productcategoryname) VALUES ('" & TextBox1.Text & "')"

            comm.CommandText = query
            comm.Connection = conn
            comm.ExecuteNonQuery()

            MsgBox("Saved")

        Catch ex As Exception
            MsgBox(ex.ToString)
        Finally
            conn.Close()
        End Try

        Try
            conn.ConnectionString = connstring
            conn.Open()

            Dim comm As New MySqlCommand
            Dim query As String = "INSERT INTO logstbl(username,process,description,dateandtime) VALUES('" & LoginForm.username & "','Inventory - New Category','New category added (" & TextBox1.Text & ")',NOW())"

            comm.Connection = conn
            comm.CommandText = query
            comm.ExecuteNonQuery()
        Catch ex As Exception
            MsgBox(ex.ToString)
        Finally
            conn.Close()
        End Try


        Me.Close()
        InventoryEditCategoryForm.refreshcategory()
    End Sub

    Private Sub Button2_Click(sender As Object, e As EventArgs) Handles Button2.Click
        Me.Close()
    End Sub


End Class

File: ACE Hardware/ACE Hardware/InventoryEditCategoryAddForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using MySql.Data.MySqlClient;

namespace ACE_Hardware
{
    public partial class InventoryEditCategoryAddForm : Form
    {

        private MySqlConnection conn = new MySqlConnection();
        private string connstring = Program.connstring;


        public InventoryEditCategoryAddForm()
        {
            InitializeComponent();
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            if (TextBox1.Text == "")
            {
                MessageBox.Show("Please input name for new category");
                return;
            }
            if (Program.inventoryEditCategoryFormInstance.ListBox1.Items.Contains(this.TextBox1.Text))
            {
                MessageBox.Show("Category already exists");
                return;
            }
            try
            {
                conn.ConnectionString = connstring;
                conn.Open();

                MySqlCommand comm = new MySqlCommand();
                string query = "INSERT INTO productcategorytbl (productcategoryname) VALUES ('" + TextBox1.Text + "')";

                comm.CommandText = query;
                comm.Connection = conn;
                comm.ExecuteNonQuery();

                MessageBox.Show("Saved");
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());
            }
            finally
            {
                conn.Close();
            }

            try
            {
                conn.ConnectionString = connstring;
                conn.Open();

                MySqlCommand comm = new MySqlCommand();
                string query = "INSERT INTO logstbl(username,process,description,dateandtime) VALUES('" + Program.login.username + "','Inventory - New Category','New category added (" + TextBox1.Text + ")',NOW())";

                comm.Connection = conn;
                comm.CommandText = query;
                comm.ExecuteNonQuery();
            }
            catch (Exception ex)
            {
                MessageBox.Show(ex.ToString());
            }
            finally
            {
                conn.Close();
            }


            this.Close();
            Program.inventoryEditCategoryFormInstance.refreshcategory();
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            this.Close();
        }
    }
}
